storage/innobase/btr/btr0cur.cc

```diff
@@ -852,6 +852,10 @@ btr_cur_search_to_nth_level(
 		}
 
 		goto search_loop;
+	} else {
+		if (level == 0) {
+			srv_stats.reset_scan_stats();
```

Comment:
The counter is global and not session and so the counter values will be corrupted under any kind of concurrency right?

Reply:
Reworked. Counters are now stored in `THD::status_var`